File: src/main/java/com/hobojoe/enderore/item/ItemBase.java
package com.hobojoe.enderore.item;

import com.hobojoe.enderore.EnderOre;
import net.minecraft.creativetab.CreativeTabs;
import net.minecraft.item.Item;

/**
 * Created by Joseph on 11/18/2016.
 */
public class ItemBase extends Item {

    protected String name;

    public ItemBase(String name) {
        this.name = name;
        setUnlocalizedName(name);
        setRegistryName(name);
    }

    public void registerItemModel() {
        EnderOre.proxy.registerItemRenderer(this, 0, name);
    }

    @Override
    public ItemBase setCreativeTab(CreativeTabs tab) {
        super.setCreativeTab(tab);
        return this;
    }

}

File: src/main/java/com/hobojoe/enderore/item/ItemBase.kt
package com.hobojoe.enderore.item

import com.hobojoe.enderore.EnderOre
import net.minecraft.creativetab.CreativeTabs
import net.minecraft.item.Item

/**
 * Created by Joseph on 11/18/2016.
 */
class ItemBase(private var name: String) : Item() {

    init {
        unlocalizedName = name
        setRegistryName(name)
    }

    fun registerItemModel() {
        EnderOre.proxy?.registerItemRenderer(this, 0, name)
    }

    override fun setCreativeTab(tab: CreativeTabs): ItemBase {
        super.setCreativeTab(tab)
        return this
    }
}